pkg/connector/client.go

```diff
@@ -63,17 +64,50 @@ func (s *SlackConnector) LoadUserLogin(ctx context.Context, login *bridgev2.User
 	teamID, userID := slackid.ParseUserLoginID(login.ID)
 	meta := login.Metadata.(*slackid.UserLoginMetadata)
 	var sc *SlackClient
+
 	if meta.Token == "" {
 		sc = &SlackClient{Main: s, UserLogin: login, UserID: userID, TeamID: teamID}
 	} else {
-		client := makeSlackClient(&login.Log, meta.Token, meta.CookieToken, meta.AppToken)
+		token := meta.Token
+		cookieToken := meta.CookieToken
+
+		if meta.Token == EphemeralTokenValueV0 {
+			var found bool
+			token, cookieToken, found = mt.Retrieve(login.ID)
+			if !found {
+				roomID, err := login.User.GetManagementRoom(ctx)
+				if err != nil {
+					return err
+				}
+
+				msgText := `The server has forgotten your login for <strong>` + string(login.ID) + `</strong>, as requested, probably because it needed to restart.
+<br/><br/>
+I, the bot, left you a &#9757 command above &#9757 to help you log back in to this workspace easily.
+<br/><br/>
+If you want to reconnect, please copy the command and paste it into the chat with me here.`
```

Comment:
Is there a reason the bot can't just read its own chat history to get the contents of the command without user intervention?

Reply:
I didn't find that capability defined in existing code; perhaps it's possible. I can re-investigate now I know my way around a bit better

Reply:
hmm I found a client method to list messages, might be doabe if I can plumb it correctly, though it may complicate things in the corner cases

Reply:
yeah looks like that'll require `mautrix/go` changes to include messages in the Matrix API, then consume it in `mautrix/slack`

Reply:
Hmm actually ...would this mean that this isn't actually a secure way to forget credentials? i.e. if the bot can read these credentials after it's rebooted, presumably any of us could too

Reply:
like maybe it's practically challenging cuz it would require code changes, but is it technically possible given access to the DB dump?

Reply:
like maybe it's practically challenging cuz it would require code changes, but is it technically possible given access to the DB dump?

Reply:
right, with the bot's keys and code changes, we'd need to be concerned about a lot more than plantext tokens stored in the DB